lib/hurley/client.rb

```diff
@@ -176,40 +176,53 @@ def inspect
     end
 
     def prepare!
-      if value = !header[:authorization] && url.basic_auth
-        header[:authorization] = value
-      end
+      prepare_basic_auth!
 
       if body
-        ctype = nil
-        case body
-        when Query
-          ctype, io = body.to_form
-          self.body = io
-        when Hash
-          ctype, io = options.build_form(body)
-          self.body = io
-        end
-        header[:content_type] ||= ctype || DEFAULT_TYPE
+        prepare_body!
       else
         return unless REQUIRED_BODY_VERBS.include?(verb)
       end
 
-      if !header.key?(:content_length) && header[:transfer_encoding] != CHUNKED
-        if body
-          if sizer = SIZE_METHODS.detect { |method| body.respond_to?(method) }
-            header[:content_length] = body.send(sizer).to_i
-          else
-            header[:transfer_encoding] = CHUNKED
-          end
+      prepare_content_length!
+    end
+
+    private
+
+    def prepare_basic_auth!
+      value = !header[:authorization] && url.basic_auth
+      header[:authorization] = value if value
+    end
+
+    def prepare_body!
+      ctype = nil
+      case body
+      when Query
+        ctype, io = body.to_form
+        self.body = io
+      when Hash
+        ctype, io = options.build_form(body)
+        self.body = io
+      end
+      header[:content_type] ||= ctype || DEFAULT_TYPE
+    end
+
+    def prepare_content_length!
+      if header.key?(:content_length) || header[:transfer_encoding] == CHUNKED
```

Comment:
> why not make this look more like a guard clause?

That line is long enough as it is.

> why not be more descriptive? 

Not sure it's needed.  I do plan to go through and write Tomdoc comments for all the methods when we're done.  Just waiting for the dust to settle a bit with this v0.1 library.  

> I also wonder if prepare_body! is doing too much?

Body and content-type are pretty closely related.  Especially when you're converting some ruby object into a body.  I thought about making teaching `RequestOptions#build_form` how to handle Query objects.  All it's doing is converting a Hash to the Query type (Query::Nested or Query::Flat) the Request is using, and calling `#to_form` on that.  I also thought about renaming `#to_form` to something generic like `#prepare_body_and_type`.  Then Hurley can turn any ruby object into a body as long as it defines that method.  That could come from a future PR.